datafusion/sqllogictest/test_files/tpch/q2.slt.part

```diff
@@ -123,64 +123,61 @@ physical_plan
 22)------------------------------------------HashJoinExec: mode=Partitioned, join_type=Inner, on=[(p_partkey@0, ps_partkey@0)], projection=[p_partkey@0, p_mfgr@1, ps_suppkey@3, ps_supplycost@4]
 23)--------------------------------------------CoalesceBatchesExec: target_batch_size=8192
 24)----------------------------------------------RepartitionExec: partitioning=Hash([p_partkey@0], 4), input_partitions=4
-25)------------------------------------------------ProjectionExec: expr=[p_partkey@0 as p_partkey, p_mfgr@1 as p_mfgr]
```

Comment:
these plans are doubly good -- they save two copies -- one in `FilterExec` and one in `CoalesceBatchesExec`